src/content/en/fundamentals/instant-and-offline/pwa-architecture/index.md

```diff
@@ -0,0 +1,922 @@
+
+### Components {: #components }
+
+<table class="details">
+  <tr>
+    <th>Component</th>
+    <th>Description</th>
+  </tr>
+  <tr>
+    <td>app shell</td>
+    <td>The minimal HTML, CSS, and JavaScript and any other static resources that provide the structure for your page, minus the actual content specific to the page. </td>
+  </tr>
+  <tr>
+    <td>cache</td>
+    <td>There are two types of cache: the automatic browser cache and program-controlled caches (<a href="https://developer.mozilla.org/en-US/docs/Web/API/Cache">Cache API</a> and <a href="https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API">IndexedDB</a>). 
+The browser cache is a temporary storage location on your computer for files downloaded by your browser to display websites. Files that are cached locally include any documents that make up a website, such as HTML files, CSS style sheets, JavaScript scripts, as well as graphic images and other multimedia content.
+The service worker also creates cache independent of the browser cache using the Cache Storage API or IndexedDB. The cache objects hold the same kinds of assets as a browser cache but make them offline accessible.The service worker provides these to enable offline support in browsers. This is referred to as "Cache (Storage) API" in this document.
+
+The cache objects are one tool you can use when building your app, but you must use them appropriately for each resource. Several caching strategies are described in Determine the Best Caching Strategy. </td>
+  </tr>
+  <tr>
+    <td>client-side rendering (CSR)</td>
+    <td>Client-side rendering means JavaScript running in the browser produces HTML (probably via templating). The benefit is you can update the screen instantly when the user clicks, rather than waiting a few hundred milliseconds at least while the server is contacted to ask what to display. Sites where you mostly navigate and view static content can get away with mostly server-side rendering. Any portion of a page that is animated or highly interactive (a draggable slider, a sortable table, a dropdown menu) almost certainly uses client-side rendering.</td>
+  </tr>
+  <tr>
+    <td>dynamic content
+</td>
+    <td>Dynamic content is all of the data, images, and other resources that your web app needs to function, but exists independently from your app shell. Although the app shell is intended to quickly populate the content of your site, users might expect dynamic content, in which case your app must fetch data specific to the user’s needs. Sometimes an app pulls this data from external, third-party APIs, and sometimes from first-party data that is dynamically generated or frequently updated.</td>
```

Comment:
'needs to function' sounds too much like the purpose of the app shell. Isn't this really the data/features users have come to see?